src/melee.cpp

```diff
@@ -1642,31 +1642,32 @@ bool Character::block_hit( Creature *source, bodypart_id &bp_hit, damage_instanc
 
         handle_melee_wear( shield, wear_modifier );
     } else {
-        // Choose which body part to block with, assume left side first
-        if( martial_arts_data->can_leg_block( *this ) && martial_arts_data->can_arm_block( *this ) ) {
-            bp_hit = one_in( 2 ) ? bodypart_id( "leg_l" ) : bodypart_id( "arm_l" );
-        } else if( martial_arts_data->can_leg_block( *this ) ) {
-            bp_hit = bodypart_id( "leg_l" );
-        } else {
-            bp_hit = bodypart_id( "arm_l" );
-        }
-
-        // Check if we should actually use the right side to block
-        if( bp_hit == bodypart_id( "leg_l" ) ) {
-            if( get_part_hp_cur( bodypart_id( "leg_r" ) ) > get_part_hp_cur( bodypart_id( "leg_l" ) ) ) {
-                bp_hit = bodypart_id( "leg_r" );
-            }
-        } else {
-            if( get_part_hp_cur( bodypart_id( "arm_r" ) ) > get_part_hp_cur( bodypart_id( "arm_l" ) ) ) {
-                bp_hit = bodypart_id( "arm_r" );
-            }
-        }
-
-        // At this point, we know we won't try blocking with items, only with limbs.
-        // But there are no limbs left, so we can disable further attempts at blocking.
-        if( get_part_hp_cur( bp_hit ) <= 0 ) {
+        std::vector<bodypart_id> block_parts;
+        if( martial_arts_data->can_leg_block( *this ) ) {
+            block_parts.push_back( bodypart_id( "leg_l" ) );
+            block_parts.push_back( bodypart_id( "leg_r" ) );
+        }
+        // If you have no martial arts you can still try to block with your arms.
+        // But martial arts with leg blocks only don't magically get arm blocks.
+        // Edge case: Leg block only martial arts gain arm blocks if both legs broken.
+        if( martial_arts_data->can_arm_block( *this ) || block_parts.empty() ) {
+            block_parts.push_back( bodypart_id( "arm_l" ) );
+            block_parts.push_back( bodypart_id( "arm_r" ) );
+        }
+        block_parts.erase( std::remove_if( block_parts.begin(),
+        block_parts.end(), [this]( bodypart_id & bpid ) {
+            return get_part_hp_cur( bpid ) <= 0;
+        } ), block_parts.end() );
+
+        if( block_parts.empty() ) {
+            // We have no parts with HP to block with.
             blocks_left = 0;
             return false;
+        } else {
+            std::sort( block_parts.begin(), block_parts.end(), [this]( bodypart_id & lhs, bodypart_id & rhs ) {
+                return get_part_hp_cur( lhs ) > get_part_hp_cur( rhs );
+            } );
+            bp_hit = block_parts[0];
         }
```

Comment:
```suggestion
        const auto part_hp_cmp = [this]( const bodypart_id & lhs, const bodypart_id & rhs ) {
            return get_part_hp_cur( lhs ) < get_part_hp_cur( rhs );
        };
        auto healthiest = std::max_element( block_parts.begin(), block_parts.end(), part_hp_cmp );
        if( healthiest == block_parts.end() ) {
            // We have no parts with HP to block with.
            return false;
        }
        bp_hit = *healthiest;
```
it's a small vector but using [max_element](https://en.cppreference.com/w/cpp/algorithm/max_element) would deliver the intent better here.

Reply:
Was wondering if there was a better way to do this.

Reply:
Something weird about this implementation. The prior implementation had hits still strike other NPC body parts, this one still focuses on the blockable parts.